quickwit/quickwit-serve/README.md

```diff
@@ -42,3 +42,13 @@ The server also exposes at `/ui` all static files located in `quickwit-ui/build`
 produced by the react app build in `quickwit-ui`.
 During development, the server will serve the local files. When building the binary, these static files will be embedded in it.
 
+## Elastic Compatible API
+
+To help implement Elasticsearch compatible API in Quickwit, we generate endpoints and related types using the [elastic-api-generator](./src/elastic_api_generator.rs) binary.
+
+To update the generated file, you will have to: 
```

Comment:
Let's automate this further as part of a `build.rs` script and remove the bin:
1. If the spec file does not exist, download it. Otherwise, issue a HEAD request to compare the length and/or modification date and download if necessary (bonus point for conditional GET)
2. Run code generation if we detect a change.
3. Disable by default but run in CI on release builds
4. Add the ability to opt-in via.. env var maybe?

Reply:
I am ok with doing the above if you feel it's still useful now.
However, my take is that this is not a good candidate for automated build.rs at least right now that we are not managing versions. We are currently working based on a released version of the Elastic stack. Meaning these specs files won't change unless we switch versions. 
I would think of this type of automation if we were managing different versions or grabbing the latest from the repository. 
This script is meant to be intentionally run when you want to support more Elasticsearch endpoints in Quickwit. 
I don't see the point of generating it from build.rs step and cargo-formating it if it's not going to change. The only time it changes is when we include a new endpoint spec file which should be intentional. Because this will require writing and registering the handlers in warp server.